engines/suppliers/app/presenters/suppliers/producer_presenter.rb

```diff
@@ -0,0 +1,44 @@
+module Suppliers
+  class ProducerPresenter
+    # @param producer [Producer]
+    # @param user [User]
+    def initialize(producer, user)
+      @producer = producer
+      @user = user
+    end
+
+    # Checks if the given user has edit rights on the producer
+    #
+    # @return [Boolean]
+    def can_edit
```

Comment:
it's an attribute, for now the serializers are just mapping attributes from models/presenters directly. Maybe an alias or something. will check.